pants_plugins/sendwave/pants_docker/package.py

```diff
@@ -223,16 +220,36 @@ async def package_into_image(
     process_path = docker_paths.first_path.path
     # build an list of arguments of the form ["-t",
     # "registry/name:tag"] to pass to the docker executable
+    full_target_name = _build_full_target_name(
+        target_name, field_set.registry.value
+    )
     tag_arguments = _build_tag_argument_list(
-        target_name, field_set.tags.value or [], field_set.registry.value
+        full_target_name, field_set.tags.value or []
     )
     # create the image
-    process_args = [process_path, "build"]
+    # We need to enable BuildKit if we plan to read and write to a remote cache
+    cli_command = (
+        ["buildx", "build"] if docker.options.cache_from else ["build"]
+    )
+    process_args = [process_path] + cli_command
     process_args.extend(tag_arguments)
     process_args.append(".")  # use current (sealed) directory as build context
     if docker.options.report_progress:
         process_args.append("--progress")
         process_args.append("plain")
+    if docker.options.cache_from:
+        registry_separator = (
+            "" if docker.options.cache_from.endswith("/") else "/"
+        )
+        cache_registry = (
+            f"{docker.options.cache_from}{registry_separator}{full_target_name}"
```

Comment:
it might be simpler to have the full image be value of the cache_from field? since we still need to put in the registry anyway so it doesn't seem like it saves us very much configuration to not include the image name, and it does add some cognitive overhead about what the plugin is doing!

(up to you though, either way works)

Reply:
Given that my initial design had `cache_from` configured as a subsystem option, it seemed to me that we needed to leave the image name out of the config value in order to allow caching to work for multiple images. But I agree with you that we should move that config value to a field on the `docker` target, so I will indeed change it to include the image name!

Reply:
yay thanks!